### PR TITLE
feat(synthetics): *revised* changes to disallow usage of the legacy runtime with new monitors starting August 26, 2024

### PR DESCRIPTION
**Pull request by pranav-new-relic**
#### Summary
This PR has been opened as an alternative to #2727, to include changes to reflect the changes to be added to the Synthetics API, for the first phase of the EOL of the Synthetics Legacy Runtime on August 26, 2024.

This includes the following changes - 

#### (A) Code
- addition of validation functions on the schema of all Synthetic Monitor resources
- these validation functions **intend to block requests to create new monitors** if they:
   - do not include fields `runtime_type` and `runtime_type_version`, or include these attributes with values as empty strings (effectively meaning, the request sent to NerdGraph includes empty runtime attributes)
   - include fields `runtime_type` and `runtime_type_version` but with values of the legacy runtime (`CHROME_BROWSER` `100`, `NODE_API` `10`)
- these validation functions **do not block requests from monitors already created**, as the EOL on August 26 is only for creation of new monitors
- however, since there could be the possibility of a small set of enterprise customers requesting for exemptions to continue to create monitors in the legacy runtime until the October 22 EOL, these changes also introduce a new attribute `use_unsupported_legacy_runtime`  (which shall be `false` by default, and an optional attribute). If set `true` by the customer, this would help skip validation functions written to block operations to create new monitors in the legacy runtime.

#### (B) Tests
- addition of a new test file, `structures_newrelic_synthetics_all_monitors_validation_helpers_test.go` which covers all scenarios associated with the errors thrown when the configuration does not include runtime attributes, or includes them with values which do not represent the new runtime (and also cases when `use_unsupported_legacy_runtime` is added to the configuration, which leads to the validation error being bypassed) 
- updates to integration tests across all Synthetic Monitor resources, as a huge lot of these tests do not have runtime attributes specified and are therefore, running into the validation error we've added; these have instead been moved to the new runtime, since test cases with the legacy runtime of all kinds of monitors are now covered by  `structures_newrelic_synthetics_all_monitors_validation_helpers_test.go`. No Synthetics tests are breaking after all these changes.

#### (C) Documentation
- updates to warnings in the docs of all five Synthetic Monitor resources with the release number of the release to go out on August 26, 2024, to make clear that changes to restrict new monitors from being created in the legacy runtime are now live via the Synthetics API, and via the New Relic Terraform Provider starting v3.43.0
- changes similar to the aforementioned in the Synthetics Legacy Runtime EOL Guide

### CODE REVIEW

File: newrelic/resource_helpers_synthetics.go
Comment (pranav-new-relic) [inline newrelic/resource_helpers_synthetics.go:1]: `resource_helpers_synthetics.go`
- includes the schema definition of the new attribute, `use_legacy_runtime_unsupported`
- includes constants which comprise values of attribute names, legacy and new runtime versions to be reused across our new validation checks, integration tests and more

File: newrelic/resource_newrelic_synthetics_broken_links_monitor.go
Comment (pranav-new-relic) [inline newrelic/resource_newrelic_synthetics_broken_links_monitor.go:1]: `resource_newrelic_synthetics_broken_links_monitor.go`
- adds `use_legacy_runtime_unsupported` to the schema of the resource
- adds the validation function which would display an error on `terraform plan` if no runtime attributes are specified in the configuratopn

File: newrelic/resource_newrelic_synthetics_cert_check_monitor.go
Comment (pranav-new-relic) [inline newrelic/resource_newrelic_synthetics_cert_check_monitor.go:1]: `resource_newrelic_synthetics_cert_check_monitor.go`
- adds `use_legacy_runtime_unsupported` to the schema of the resource
- adds the validation function which would display an error on `terraform plan` if no runtime attributes are specified in the configuration

File: newrelic/resource_newrelic_synthetics_monitor.go
Comment (pranav-new-relic) [inline newrelic/resource_newrelic_synthetics_monitor.go:1]: `resource_newrelic_synthetics_monitor.go`
- adds `use_legacy_runtime_unsupported` to the schema of the resource
- adds the validation function which would display an error on `terraform plan` if no runtime attributes are specified in the configuration

File: newrelic/resource_newrelic_synthetics_script_monitor.go
Comment (pranav-new-relic) [inline newrelic/resource_newrelic_synthetics_script_monitor.go:1]: `resource_newrelic_synthetics_script_monitor.go`
- adds `use_legacy_runtime_unsupported` to the schema of the resource
- adds the validation function which would display an error on `terraform plan` if no runtime attributes are specified in the configuration

File: newrelic/resource_newrelic_synthetics_step_monitor.go
Comment (pranav-new-relic) [inline newrelic/resource_newrelic_synthetics_step_monitor.go:1]: `resource_newrelic_synthetics_step_monitor.go`
- adds `use_legacy_runtime_unsupported` to the schema of the resource
- adds the validation function which would display an error on `terraform plan` if no runtime attributes are specified in the configuration

**Review by sanderblue — DISMISSED** [review]

**Pull request merged**